### PR TITLE
[fix] PAN-304 error handling when requested entity is not exist

### PR DESCRIPTION
**Pull request by denysov-serhii**

### CODE REVIEW

File: Mappers/DbMapper.php [hunk 1]
```diff
@@ -199,7 +202,13 @@ public function get(array $identifiers) : ArraySerializable
             $conditions[] = new EqualCriteria($indName, $indValue);
         }
 
-        return $this->load(new SearchCriteriaConstructor($conditions, 1))->getFirst();
+        $entity = $this->load(new SearchCriteriaConstructor($conditions, 1))->getFirst();
+
+        if (is_null($entity)) {
```
Comment (phpadventure) [inline Mappers/DbMapper.php]: It's bad manner to check on null object, use collection count instead

File: Exceptions/EntityNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Infrastructure\Exceptions;
+
+
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+
+class EntityNotFoundException extends ResourceNotFoundException
+{
+
+}
```
Comment (phpadventure) [inline Exceptions/EntityNotFoundException.php]: Why you don't use this exception "HttpResourceNotFoundException" ?

File: Mappers/DbMapper.php [hunk 3]
```diff
@@ -199,7 +202,13 @@ public function get(array $identifiers) : ArraySerializable
...
+            throw new EntityNotFoundException();
```
Comment (phpadventure) [inline Mappers/DbMapper.php]: This exception will end up with 500 status code

Comment (denysov-serhii) [reply Mappers/DbMapper.php]: This exception will end up with 404 status code, pls see on Application class

File: Exceptions/EntityNotFoundException.php (hunk 2 above)
Comment (denysov-serhii) [reply Exceptions/EntityNotFoundException.php]: because it is not a http mapper

File: Mappers/DbMapper.php (hunk 3 above)
Comment (phpadventure) [reply Mappers/DbMapper.php]: ok

**Pull request merged**